<a href="https://colab.research.google.com/github/valesed/GitHub-Repository-for-CPE311/blob/main/Seatwork_6_2_Programming_Exercise_Getting_Started_with_Pandas!_Calatraba.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **sing the 2019_Yellow_Taxi_Trip_Data.csv Links to an external site. dataset, accomplish the following items and submit a PDF of the notebook:

1. Create a DataFrame by reading in the 2019_Yellow_Taxi_Trip_Data.csv file. Examine the first 5 rows.
2. Find the dimensions (number of rows and number of columns) in the data.

3. Using the data in the 2019_Yellow_Taxi_Trip_Data.csv file, calculate summary statistics for the fare_amount, tip_amount, tolls_amount, and total_amount columns.

4. Isolate the fare_amount, tip_amount, tolls_amount, and total_amount for the longest trip by distance (trip_distance).

**

In [ ]:
import pandas as pd

data = pd.read_csv("/content/2019_Yellow_Taxi_Trip_Data.csv")

#Create a DataFrame by reading in the 2019_Yellow_Taxi_Trip_Data.csv file.
data.head()

,vendorid,tpep_pickup_datetime,tpep_dropoff_datetime,passenger_count,trip_distance,ratecodeid,store_and_fwd_flag,pulocationid,dolocationid,payment_type,fare_amount,extra,mta_tax,tip_amount,tolls_amount,improvement_surcharge,total_amount,congestion_surcharge
0,2,2019-10-23T16:39:42.000,2019-10-23T17:14:10.000,1,7.93,1,N,138,170,1,29.5,1.0,0.5,7.98,6.12,0.3,47.90,2.5
1,1,2019-10-23T16:32:08.000,2019-10-23T16:45:26.000,1,2.00,1,N,11,26,1,10.5,1.0,0.5,0.00,0.00,0.3,12.30,0.0
2,2,2019-10-23T16:08:44.000,2019-10-23T16:21:11.000,1,1.36,1,N,163,162,1,9.5,1.0,0.5,2.00,0.00,0.3,15.80,2.5
3,2,2019-10-23T16:22:44.000,2019-10-23T16:43:26.000,1,1.00,1,N,170,163,1,13.0,1.0,0.5,4.32,0.00,0.3,21.62,2.5
4,2,2019-10-23T16:45:11.000,2019-10-23T16:58:49.000,1,1.96,1,N,163,236,1,10.5,1.0,0.5,0.50,0.00,0.3,15.30,2.5


In [ ]:
#Examine the first 5 rows.
data.tail(5)

,vendorid,tpep_pickup_datetime,tpep_dropoff_datetime,passenger_count,trip_distance,ratecodeid,store_and_fwd_flag,pulocationid,dolocationid,payment_type,fare_amount,extra,mta_tax,tip_amount,tolls_amount,improvement_surcharge,total_amount,congestion_surcharge
9995,1,2019-10-23T17:39:59.000,2019-10-23T17:49:26.000,2,1.3,1,N,238,239,1,8.0,3.5,0.5,2.46,0.0,0.3,14.76,2.5
9996,1,2019-10-23T17:53:02.000,2019-10-23T18:00:45.000,1,1.4,1,N,239,166,2,8.0,3.5,0.5,0.00,0.0,0.3,12.30,2.5
9997,1,2019-10-23T17:07:16.000,2019-10-23T17:11:35.000,1,0.7,1,N,166,152,2,5.0,1.0,0.5,0.00,0.0,0.3,6.80,0.0
9998,1,2019-10-23T17:38:26.000,2019-10-23T17:49:28.000,2,2.5,1,N,151,42,1,10.0,1.0,0.5,0.00,0.0,0.3,11.80,0.0
9999,1,2019-10-23T17:22:14.000,2019-10-23T17:52:09.000,1,3.0,1,N,237,68,1,19.0,3.5,0.5,2.50,0.0,0.3,25.80,2.5


In [ ]:
#Find the dimensions (number of rows and number of columns) in the data.
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   vendorid               10000 non-null  int64  
 1   tpep_pickup_datetime   10000 non-null  object 
 2   tpep_dropoff_datetime  10000 non-null  object 
 3   passenger_count        10000 non-null  int64  
 4   trip_distance          10000 non-null  float64
 5   ratecodeid             10000 non-null  int64  
 6   store_and_fwd_flag     10000 non-null  object 
 7   pulocationid           10000 non-null  int64  
 8   dolocationid           10000 non-null  int64  
 9   payment_type           10000 non-null  int64  
 10  fare_amount            10000 non-null  float64
 11  extra                  10000 non-null  float64
 12  mta_tax                10000 non-null  float64
 13  tip_amount             10000 non-null  float64
 14  tolls_amount           10000 non-null  float64
 15  imp

In [ ]:

selected_features = ["fare_amount", "tip_amount", "tolls_amount", "total_amount"]

# Calculate summary statistics for the selected features
display(data[selected_features].describe())

,fare_amount,tip_amount,tolls_amount,total_amount
count,10000.000000,10000.000000,10000.000000,10000.000000
mean,15.106313,2.634494,0.623447,22.564659
std,13.954762,3.409800,6.437507,19.209255
min,-52.000000,0.000000,-6.120000,-65.920000
25%,7.000000,0.000000,0.000000,12.375000
50%,10.000000,2.000000,0.000000,16.300000
75%,16.000000,3.250000,0.000000,22.880000
max,176.000000,43.000000,612.000000,671.800000


In [ ]:
#Isolate the fare_amount, tip_amount, tolls_amount, and total_amount for the longest trip by distance (trip_distance).
longest_trip = data.sort_values(by='trip_distance', ascending=False).head(5)
display(longest_trip[selected_features])

,fare_amount,tip_amount,tolls_amount,total_amount
8338,176.0,18.29,6.12,201.21
9965,52.0,0.00,6.12,65.92
1656,52.0,13.18,6.12,79.10
2237,87.5,0.00,6.12,95.42
436,52.0,13.18,6.12,79.10


# **Conclusion**

in this activity I performed I have mostly based it on the first activity(6.1) for most of the codes used though especially the last 2 parts nonetheless it was an enjoyable experience